src/api/db/session.ts

```diff
@@ -1,13 +1,114 @@
-import { database } from "../firebase";
+import { database, getSessionRef, getUniqueId } from "../firebase";
+import {
+  Normalized,
+  Session,
+  NoGameSession,
+  SessionWithId,
+  DocumentSnapshot,
+  QuerySnapshot,
+} from "../../model/Session";
+import { Player, PlayerWithId } from "../../model/Player";
+import { buildOne } from "../../model/Card";
 
-export async function requestCreateSession(name: string) {
-  try {
-    console.log("set");
-    const res = await database.collection("session").doc().set({
-      name: "david",
+const codeGenerator = () => {
+  return String(Math.round(Math.random() * 100000));
+};
+
+function normalizeDocument<T>(doc: DocumentSnapshot): Normalized<T> {
+  if (doc.exists) {
+    const data = doc.data() as T;
+    return {
+      [doc.id]: data,
+    };
+  } else {
+    return {};
+  }
+}
+
+function normalizeQuery<T>(doc: QuerySnapshot): Normalized<T> {
+  if (!doc.empty) {
+    let data = {};
+    doc.forEach((el) => {
+      data = {
+        ...data,
+        [el.id]: el.data(),
+      };
     });
-    console.log(res);
-  } catch (error) {
-    console.error(error);
+    return data;
+  } else {
+    return {};
   }
 }
+
+//TODO: IMPROVE THIS CODE
+export async function requestCreateSession(
```

Comment:
Put a Try/Catch on it's contents, none of the functions has one, we could have at least one in this top one.

Reply:
This is done in the redux actions =)